askama_derive/src/generator.rs

```diff
@@ -719,11 +719,15 @@ impl<'a> Generator<'a> {
         let mut names = Buffer::new(0);
         let mut values = Buffer::new(0);
         let mut is_first_variable = true;
-        for (i, arg) in def.args.iter().enumerate() {
-            let expr = args.get(i).ok_or_else(|| {
-                CompileError::from(format!("macro {name:?} takes more than {i} arguments"))
-            })?;
-
+        if args.len() != def.args.len() {
+            return Err(CompileError::from(format!(
+                "macro {name:?} expected {} argument{}, found {}",
+                def.args.len(),
+                if def.args.len() > 1 { "s" } else { "" },
+                args.len()
+            )));
+        }
+        for (expr, arg) in std::iter::zip(args, &def.args) {
```

Comment:
Would you mind extracting the `enumerate()` + `.get()` -> `zip()` change into a separate commit, please?

Reply:
Sure.